### PR TITLE
Introduce source_type flag

### PR DESCRIPTION
**Pull request by pacospace**
Signed-off-by: Francesco Murdaca <fmurdaca@redhat.com>

### CODE REVIEW

**Comment by pacospace** [comment]
> LGTM

I have a question for the scrum.

File: thamos/cli.py [hunk 1]
```diff
@@ -381,6 +382,7 @@ def advise(
             debug=debug,
             nowait=no_wait,
             force=force,
+            source_type=ThothAdviserIntegrationEnum.CLI.value
```
Comment (fridex) [inline thamos/cli.py:385]: BTW this should be:
```python
            source_type=ThothAdviserIntegrationEnum.CLI
```

to conform signature:

```python
    source_type: typing.Optional[ThothAdviserIntegrationEnum] = None,
```

And in the advise method:

```python
        "source_type": source_type.name,

```

Comment (pacospace) [reply thamos/cli.py]: Updated and also added check on s2i using the method.

File: thamos/lib.py [hunk 2]
```diff
@@ -321,11 +323,14 @@ def advise(
     if recommendation_type:
         recommendation_type = recommendation_type.lower()
 
+    if _is_s2i():
```
Comment (fridex) [inline thamos/lib.py:326]: Maybe:

```python
if source_type is None and _is_s2i():
        source_type = ThothAdviserIntegrationEnum.S2I
```

.. not to override parameters, these are usually hard to debug later on.

**Comment by pacospace** [comment]
recheck

**Pull request merged**